python/ray/train/_internal/storage.py

```diff
@@ -614,37 +590,85 @@ def experiment_fs_path(self) -> str:
+        See `trial_staging_path` for files that should be written to trial folders.
+
+        The directory is synced to
+        `{storage_path}/{experiment_dir_name}` periodically.
+        See `_ExperimentCheckpointManager.checkpoint` for where that happens.
         """
-        return Path(self.storage_local_path, self.experiment_dir_name).as_posix()
+        return Path(
+            get_ray_train_session_dir(), self.experiment_dir_name, "driver_artifacts"
+        ).as_posix()
 
     @property
-    def trial_local_path(self) -> str:
-        """The local filesystem path to the trial directory.
+    def trial_fs_path(self) -> str:
+        """The trial directory path on the `storage_filesystem`.
 
         Raises a ValueError if `trial_dir_name` is not set beforehand.
         """
         if self.trial_dir_name is None:
             raise RuntimeError(
-                "Should not access `trial_local_path` without setting `trial_dir_name`"
+                "Should not access `trial_fs_path` without setting `trial_dir_name`"
             )
-        return Path(self.experiment_local_path, self.trial_dir_name).as_posix()
+        return Path(self.experiment_fs_path, self.trial_dir_name).as_posix()
 
     @property
-    def trial_fs_path(self) -> str:
-        """The trial directory path on the `storage_filesystem`.
+    def trial_local_staging_path(self) -> str:
+        """The local filesystem path of the trial directory on the driver.
 
-        Raises a ValueError if `trial_dir_name` is not set beforehand.
+        The driver is the node where `Trainer.fit`/`Tuner.fit` is being called.
+
+        This path is of the form:
+        `/tmp/ray/session_<session_id>/artifacts/<experiment_dir_name>/driver_artifacts/<trial_dir_name>`
+
+        This should be used as the temporary location for files on the driver
+        before syncing them to `(storage_filesystem, storage_path)`.
+
+        For example, callbacks (e.g., JsonLoggerCallback) should write trial-specific
+        logfiles within this directory.
         """
         if self.trial_dir_name is None:
             raise RuntimeError(
-                "Should not access `trial_fs_path` without setting `trial_dir_name`"
+                "Should not access `trial_local_staging_path` "
+                "without setting `trial_dir_name`"
             )
-        return Path(self.experiment_fs_path, self.trial_dir_name).as_posix()
+        return Path(self.experiment_local_staging_path, self.trial_dir_name).as_posix()
+
+    @property
+    def trial_working_directory(self) -> str:
+        """The local filesystem path to trial working directory.
+
+        This path is of the form:
+        `/tmp/ray/session_<session_id>/artifacts/<experiment_dir_name>/working_dirs/<trial_dir_name>`
+
+        Ray Train/Tune moves the remote actor's working directory to this path
+        by default, unless disabled by `RAY_CHDIR_TO_TRIAL_DIR` environment variable.
```

Comment:
Mention the result trial dir abs path?
`{RAY_CHDIR_TO_TRIAL_DIR}/working_dirs/<trial_dir_name>` or `{RAY_CHDIR_TO_TRIAL_DIR}/<trial_dir_name>`?